bitsandbytes/autograd/_functions.py

```diff
@@ -366,10 +357,24 @@ def backward(ctx, grad_output):
         if req_gradA:
             C32grad, Sgrad = F.transform(Cgrad, "col32")
             if state.CxBt is None:
-                state.CxBt, state.SBt = F.transform(
-                    state.CBt, to_order=formatB, transpose=True
+                if state.has_fp16_weights:
+                    CBt = state.CBt
+                else:
+                    # Restore CBt from CB
+                    assert state.CBt is None, "CBt should not be stored in state"
+                    CB = state.CB.half()
+                    SCB = state.SCB.unsqueeze(1).half()
+                    SCBt = state.SCBt.unsqueeze(1).half()
+                    Bt = (CB * SCB).t().contiguous()
+                    CBt = (Bt / SCBt).t().to(torch.int8)
```

Comment:
[opinion] this block of code looks like the most memory-intensive; perhaps we can optimize some of it with in-place divisions / transpositions (please ping me if you're interested)